spinn_machine/machine.py

```diff
@@ -783,22 +783,49 @@ def spinnaker_links(self):
         return iter(self._spinnaker_links.items())
 
     def get_spinnaker_link_with_id(
-            self, spinnaker_link_id, board_address=None):
+            self, spinnaker_link_id, board_address=None, chip_coords=None):
         """ Get a SpiNNaker link with a given ID
 
         :param int spinnaker_link_id: The ID of the link
         :param board_address:
-            the board address that this SpiNNaker link is associated with
+            optional board address that this SpiNNaker link is associated with.
+            If both this and chip_coords are specified, chip_coords will be
+            used first, with board_address only used if this is unsuccessful.
         :type board_address: str or None
+        :param chip_coords:
+            optional chip coordinates that this SpiNNaker link is associated
+            with.  If both this and board_address are specified, chip_coords
+            will be used first, with board_address only used if this is
+            unsuccessful.
+        :type chip_coords: tuple(int, int) or None
         :return: The SpiNNaker link data or None if no link
-        :rtype: ~spinn_machine.link_data_objects.SpinnakerLinkData or None
-        """
+        :rtype: ~spinn_machine.link_data_objects.SpinnakerLinkData
+        """
+        # Try chip coordinates first
+        if chip_coords is not None:
+            if chip_coords not in self._chips:
+                raise KeyError(f"No chip {chip_coords} found!")
+            key = (chip_coords, spinnaker_link_id)
+            link_data = self._spinnaker_links.get(key, None)
+            if link_data is not None:
+                return link_data
+            if board_address is None:
```

Comment:
This is a little weird.

Would it not be better to say if chip_coords fails go boom?

If kept there should be a log warning here!

Also weird that if chip_coords there must be a board_address while if there is neither it works!

Reply:
I have changed this so that chip_coords just simply overrides board_address (with warning) and making board_address of None documented to mean boot board.